module/motion/QuinticWalk/src/QuinticWalk.cpp

```diff
@@ -140,30 +140,31 @@ namespace motion {
             current_orders = orders;
         });
 
-        on<Trigger<Sensors>>().then([this](const Sensors& sensors) {
-            if (config.imu_active) {
-                Eigen::Vector3f RPY =
-                    utility::math::euler::MatrixToEulerIntrinsic(sensors.Htw.topLeftCorner<3, 3>().cast<float>());
-
-                // compute the pitch offset to the currently wanted pitch of the engine
-                float wanted_pitch =
-                    params.trunk_pitch + params.trunk_pitch_p_coef_forward * walk_engine.getFootstep().getNext().x()
-                    + params.trunk_pitch_p_coef_turn * std::abs(walk_engine.getFootstep().getNext().z());
-                RPY.y() += wanted_pitch;
-
-                // threshold pitch and roll
-                if (std::abs(RPY[0]) > config.imu_roll_threshold) {
-                    log<NUClear::WARN>(fmt::format("Robot roll exceeds threshold - {} > {}",
-                                                   std::abs(RPY.x()),
-                                                   config.imu_roll_threshold));
-                    walk_engine.requestPause();
-                }
-                else if (std::abs(RPY[1]) > config.imu_pitch_threshold) {
-                    log<NUClear::WARN>(fmt::format("Robot pitch exceeds threshold - {} > {}",
-                                                   std::abs(RPY.y()),
-                                                   config.imu_pitch_threshold));
-                    walk_engine.requestPause();
-                }
+        imu_reaction = on<Trigger<Sensors>>().then([this](const Sensors& sensors) {
+            if (!config.imu_active) {
+                imu_reaction.disable();
+            }
```

Comment:
Move this to `on<Configuration>`

Reply:
It didn't work in `on<Configuration>` 🤷

Reply:
That's because you defined the reaction after the configuration block. So when you tried to disable it you hadn't even created the reaction yet.

Reply:
Ah that would be an issue yes